src/java/org/apache/cassandra/config/Config.java

```diff
@@ -76,6 +76,7 @@
     public volatile boolean hinted_handoff_enabled = true;
     public Set<String> hinted_handoff_disabled_datacenters = Sets.newConcurrentHashSet();
     public volatile int max_hint_window_in_ms = 3 * 3600 * 1000; // three hours
+    public volatile Integer max_hints_size_per_host_in_mb = 0; // non-positive means disabled
```

Comment:
Can we use primitive here to avoid any boxing overhead?

```suggestion
    public volatile int max_hints_size_per_host_in_mb = 0; // non-positive means disabled
```